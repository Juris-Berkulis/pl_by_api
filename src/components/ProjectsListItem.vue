<script setup lang="ts">
import type { Project } from '@/types';

interface Props {
    project: Project,
};

defineProps<Props>();
</script>

<template>
<li class="mb-2 last:mb-0 bg-sky-200 p-1">
    <p class="mb-0.5 font-bold">{{ project.status }}</p>
    <h2>{{ project.title }}</h2>
</li>
</template>

<style scoped>
</style>
