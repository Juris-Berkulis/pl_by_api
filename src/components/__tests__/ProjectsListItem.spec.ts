import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import ProjectsListItem from "@/components/ProjectsListItem.vue";
import type { Project } from "@/types";

const project: Project = {
    id: 1,
    title: 'aaa',
    status: 'CREATED',
};

describe('ProjectsListItem', () => {
    const wrapper = mount(ProjectsListItem, {
        props: {
            project,
        }
    });

    it('Snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Contains text', () => {
        expect(wrapper.text()).toContain('aaa');
        expect(wrapper.text()).toContain('CREATED');
    });
});
