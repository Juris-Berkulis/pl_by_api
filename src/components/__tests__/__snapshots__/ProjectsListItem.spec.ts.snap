// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProjectsListItem > Snapshot 1`] = `
"<li class=\\"mb-2 last:mb-0 bg-sky-200 p-1\\">
  <p class=\\"mb-0.5 font-bold\\">CREATED</p>
  <h2>aaa</h2>
</li>"
`;
